src/Stratis.Bitcoin.Features.Consensus/StakeValidator.cs

```diff
@@ -143,26 +308,78 @@ private static uint256 ToUInt256(BigInteger input)
-        // amount of coins one owns.
-        // The reason this hash is chosen is the following:
-        //   stakeModifierV2: Scrambles computation to make it very difficult to precompute future proof-of-stake.
-        //   stakingCoins.Time: Time of the coinstake UTXO. Slightly scrambles computation.
-        //   prevout.Hash: Hash of stakingCoins UTXO, to reduce the chance of nodes generating coinstake at the same time.
-        //   prevout.N: Output number of stakingCoins UTXO, to reduce the chance of nodes generating coinstake at the same time.
-        //   transactionTime: Timestamp of the coinstake transaction.
-        //   Block or transaction tx hash should not be used here as they can be generated in vast
-        //   quantities so as to generate blocks faster, degrading the system back into a proof-of-work situation.
-        //
+        /// <summary>
+        /// Returns <c>true</c> if provided UTXOs were confirmed in less then <see cref="maxDepth"/> number of blocks.
+        /// </summary>
+        private bool IsConfirmedInNPrevBlocks(UnspentOutputs utxoSet, ChainedBlock pindexFrom, long maxDepth)
+        {
+            this.logger.LogTrace("({0}:'{1}/{2}',{3}:'{4}',{5}:{6})", nameof(utxoSet), utxoSet.TransactionId, utxoSet.Height, nameof(pindexFrom), pindexFrom, nameof(maxDepth), maxDepth);
+
+            int actualDepth = pindexFrom.Height - (int)utxoSet.Height;
+            bool res = actualDepth < maxDepth;
+
+            this.logger.LogTrace("(-):{0}", res);
+            return res;
+        }
+
+        /// <summary>
+        /// Verifies transaction's signature.
+        /// </summary>
+        /// <param name="txFrom">UTXO that is spent in the transaction.</param>
+        /// <param name="txTo">Transaction.</param>
+        /// <param name="txToInN">Index of the transaction's input.</param>
+        private bool VerifySignature(UnspentOutputs txFrom, Transaction txTo, int txToInN, ScriptVerify flagScriptVerify)
+        {
+            this.logger.LogTrace("({0}:'{1}/{2}',{3}:{4},{5}:{6})", nameof(txFrom), txFrom.TransactionId, txFrom.Height, nameof(txToInN), txToInN, nameof(flagScriptVerify), flagScriptVerify);
+
+            TxIn input = txTo.Inputs[txToInN];
+
+            if (input.PrevOut.N >= txFrom._Outputs.Length)
+                return false;
+
+            if (input.PrevOut.Hash != txFrom.TransactionId)
+                return false;
+
+            TxOut output = txFrom._Outputs[input.PrevOut.N];
+
+            var txData = new PrecomputedTransactionData(txTo);
+            var checker = new TransactionChecker(txTo, txToInN, output.Value, txData);
+            var ctx = new ScriptEvaluationContext { ScriptVerify = flagScriptVerify };
+
+            bool res = ctx.VerifyScript(input.ScriptSig, output.ScriptPubKey, checker);
+            this.logger.LogTrace("(-):{0}", res);
+            return res;
+        }
+
+        /// <summary>
+        /// Stratis kernel protocol. Checks the stake kernel hash.
```

Comment:
i'd remove `Stratis kernel protocol.` it means nothing to me
and the actual summary could be little more descriptive - like `Checks that the stake kernel hash satisfies the target difficulty.`
also consider adding `<throws>` tag